### DISCUSSION TITLE
Adding support for signing a digest

### DISCUSSION THREAD

**Discussion by kopy-kat** [wevm/viem#1474] [Idea / Feature Request] [answered]
This would function like `signingKey.signDigest` in [ethers](https://docs.ethers.org/v5/api/utils/signing-key/#SigningKey). This is useful for some Solidity libraries that verify signatures that are not in EIP-191 format. I'm also happy to build a prototype if this is useful.

**Comment by zouhangwithsweet** [comment]
Something update?

**Reply by jxom** [reply, marked as answer]
It exists.

```typescript
import { sign } from 'viem/accounts'

const signature = await sign({
  hash: '0x...',
  privateKey: '0x...'
}) 
```

**Reply by dakom** [reply]
Are there any examples of using this with a `walletClient`?
